### PR TITLE
Executor CA Certs

### PR DESCRIPTION
**Pull request by Piszmog**
Be able to set custom CA certs for executor HTTP client.

## Test plan

Unit tests

### CODE REVIEW

File: enterprise/cmd/executor/internal/apiclient/baseclient.go [hunk 1]
```diff
@@ -83,8 +87,20 @@ func NewBaseClient(options BaseClientOptions) (*BaseClient, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	client := httpcli.InternalClient
+	if len(options.CACertificate) > 0 {
+		decodedCACerts, err := base64.StdEncoding.DecodeString(options.CACertificate)
+		client, err = httpcli.InternalClientFactory.Client(httpcli.NewCertPoolOpt(string(decodedCACerts)))
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/apiclient/baseclient.go:94]: Need a second pair of eyes on this to ensure this makes sense

File: doc/admin/executors/deploy_executors_binary.md [hunk 2]
```diff
@@ -80,34 +80,35 @@ mv executor /usr/local/bin
...
+| `EXECUTOR_FIRECRACKER_KERNEL_IMAGE`      | The base image containing the kernel binary to use for virtual machines.                                                                                                                                                           | `sourcegraph/ignite-kernel:5.10.135-amd64` |
+| `EXECUTOR_FIRECRACKER_SANDBOX_IMAGE`     | The OCI image for the ignite VM sandbox.                                                                                                                                                                                           | `sourcegraph/ignite:v0.10.5`               |
+| `EXECUTOR_CA_CERTIFICATE`                | The base64 encoded CA certificate to use for TLS connections to the frontend.                                                                                                                                                      | `warn`                                     |
```
Comment (danieldides) [inline doc/admin/executors/deploy_executors_binary.md:109]: The default value here looks like it was a copy-paste from the log level.

File: enterprise/cmd/executor/internal/apiclient/baseclient.go [hunk 3]
```diff
@@ -83,8 +87,20 @@ func NewBaseClient(options BaseClientOptions) (*BaseClient, error) {
...
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client = httpcli.InternalClient
+	}
```
Comment (danieldides) [inline enterprise/cmd/executor/internal/apiclient/baseclient.go:100]: I don't think you need this `else` block

File: enterprise/cmd/executor/internal/apiclient/baseclient.go (hunk 1 above)
Comment (sanderginn) [reply enterprise/cmd/executor/internal/apiclient/baseclient.go]: Looking at other uses of `NewCertPoolOpt` it looks good to me. Do we ever want to pass other opts to the client?

**Review by sanderginn — APPROVED** [review]

File: enterprise/cmd/executor/internal/apiclient/baseclient.go (hunk 3 above)
Comment (Piszmog) [reply enterprise/cmd/executor/internal/apiclient/baseclient.go]: Yep. I updated to just build up a slice of options and pass to the client factory

File: enterprise/cmd/executor/internal/apiclient/baseclient.go (hunk 1 above)
Comment (Piszmog) [reply enterprise/cmd/executor/internal/apiclient/baseclient.go]: Maybe in the future? Hard to say at the moment. I did update to build up a slice of options to pass to the client factory. So in the future, it'll be easier to add additional options

[the remark this answers is not in the document]
Comment (eseliger) [reply enterprise/cmd/executor/internal/apiclient/baseclient.go]: are you sure we don't need to append system as well? https://sourcegraph.com/github.com/sourcegraph/sourcegraph@rc/executor-ca-certs/-/blob/cmd/gitserver/server/internal/cacert/cacert.go?L12#tab=references gitserver seems to do that 🤔

[the remark this answers is not in the document]
Comment (Piszmog) [reply enterprise/cmd/executor/internal/apiclient/baseclient.go]: I'll take a look. Be nice if the docs clarified 😵‍💫

**Comment by Piszmog** [comment]
So loading the default certs is a bit convoluted. This makes me wonder,

- Is this still an Issue?
- Do we want to load the default certs?
    - Since a user may be calling an their own custom stuff, the default certs are not going to help

**Comment by eseliger** [comment]
we "solved" it for now by pointing to the internal IP instead of the public LB that uses the certificate. I think in general it makes sense to have _some_ guidance on how to solve this. We can also just document "mount them in this folder" or something - but documentation would be good I think.

**Comment by Piszmog** [comment]
Created #49930 to track this documentation

**Pull request closed without merging**